homeassistant/util/unit_system.py

```diff
@@ -100,12 +100,37 @@ def temperature(self: object, temperature: float, from_unit: str) -> float:
                                         from_unit, self.temperature_unit)
 
     def length(self: object, length: float, from_unit: str) -> float:
-        """Convert the given length to this unit system."""
+        """Convert the given length to this unit system return a float."""
+        converted = self.length_and_unit(length, from_unit, False)
```

Comment:
Just leave this the old implementation, it doesn't need any changing.

Reply:
Here you say "just leave this the old implementation" but the new implementation *is* the old implementation with the addition of the "smart" unit conversion. In another requested change you asked me to eliminate duplication. I can either copy/paste the old implementation into this method enclosure or I can overload the other method with a boolean to turn on/off the "smart" unit auto-ranging conversion(how it is now).

This method allows for backward compatibility for things that are not expecting the units to change. It simply calls the new implementation and tells it to not change the to units. This is the old implementation for all intents and purposes

Reply:
I disagree with that. The de-duplication comment earlier was about multiple lines of identical code. In this case it's 1 line and it's the shortcut of what else should be called. `auto_range` is a confusing parameter in the method signature for `length_and_unit` and I would like to see it removed.

Reply:
This comment needs to be addressed.